### ISSUE TITLE
For blocked URLs, we should do 200 status code instead of redirects

### ISSUE THREAD

**Issue by bbondy** [brave/brave-browser#2554]
We currently redirect to empty data URLs (or empty image data URLs for images). 
We should instead be responding with 200 status code because data URLs can cause CSP violations.

I believe this will help with general webcompat no a large scale.

See also:
https://chromium.googlesource.com/chromium/src/+/master/net/docs/life-of-a-url-request.md

**Comment by mrzealot** [comment]
Just mentioning this here as well: like #2931 shows, there can be problems with not blocking, too, if the script thinks that it actually managed to load something that we blocked. Possibly these are rarer than the problems solved by not blocking, I don't have numbers, just an FYI.

**Comment by iefremov** [comment]
@mrzealot Yeah, that makes sense. We are not going to quickly push this change to stable, I think we would rather treat this as an experiment.

**Comment by srirambv** [comment]
Moving issue to `0.62.x` based on https://github.com/brave/brave-core/pull/1377 set milestone. @iefremov please add `QA/Yes` or `QA/No` labels accordingly

**Comment by iefremov** [comment]
@srirambv thanks for pinging me, added QA/Yes

**Comment by btlechowski** [comment]
@iefremov Could you include a test plan in https://github.com/brave/brave-core/pull/1377? 

I have no idea how to test this one.

**Comment by iefremov** [comment]
@btlechowski 
1. Open Devtools
2. Open any website with lots of ads, e.g. https://www.foxnews.com/
3a. Before this change, in a DevTools console one can see a lot of entries like `Access to XMLHttpRequest at 'data:text/plain,'`
3b. After this change, these entries should vanish.
4. There are also no entries like `'data:text/plain,'` on a Network tab
5. Ads are blocked as expected

**Comment by btlechowski** [comment]
Thanks @iefremov !

Verification passed on 

Brave | 0.62.10 Chromium: 73.0.3683.39 (Official Build) dev (64-bit)
-- | --
Revision | cc53b0e12fcaf42e4bab8d6c23bd4fb7aae99f6c-refs/branch-heads/3683@{#413}
OS | Windows 7 Service Pack 1 Build 7601.24312

Used test plan from https://github.com/brave/brave-browser/issues/2554#issuecomment-468207371
Checked several sites.

![image](https://user-images.githubusercontent.com/34715963/53608526-658e5900-3bc3-11e9-9d9a-308e0ab22b9b.png)
